backend/queries/volunteers.js

```diff
@@ -94,7 +94,7 @@ const addVolunteer = async (user, password) => {
     const volunteer = await db.one(insertQuery, user);
 
     const promises = [];
-    user.skills.map(skillId => promises.push(db.none('INSERT INTO volunteer_skills (volunteer_id, skill_id) VALUES ($1, $2)', [user.userId, skillId])));
+    user.skills.map(skillId => promises.push(db.none('INSERT INTO volunteer_skills (volunteer_id, skill_id) VALUES ($1, $2)', [volunteer.v_id, skillId])));
     await Promise.all(promises);
 
     return volunteer;
```

Comment:
would you rather not combine all the responses into one returned object? Like:
user.skills.map(skillId => promises.push(db.one('INSERT INTO volunteer_skills (volunteer_id, skill_id) VALUES ($1, $2) RETURNING *;', [volunteer.v_id, skillId])));
const vs_responses = await Promise.all(promises);

return { volunteerData: volunteer, vsData: vs_responses };

Reply:
also I think it's forEach not map because you're not using the return to create a new array

Reply:
good catch for forEach, also the way you present it, returning both pieces of data is good but the way the fontend is build the skills are fetched each time the profile is rendered which makes it useless (the way I see it, I may be wrong) to send the skills as well as the user data